dvol_python/test_plugin.py

```diff
@@ -1,23 +1,151 @@
 """
-log of integration tests to write:
+Tests for dvol docker integration.
 
-write test_switch_branches_restarts_containers
+Assumes that:
+* we have docker running and are running as root or a user in the docker group
+* we have dvol under test installed in /usr/local/bin (as per Makefile)
+* we have dvol docker plugin installed and running
+* we are not running in parallel with ourselves
+* we have access to ports on the machine
+* it's totally cool to destroy dvol volumes with certain names (including but
+  not limited to memorydiskserver)
+* there are no dvol volumes on the machine
+"""
 
-command:
-    docker-compose up -d (in a directory with appropriate docker-compose.yml file)
-expected behaviour:
-    docker containers are started with dvol accordingly
+from twisted.trial.unittest import TestCase
+from twisted.python.filepath import FilePath
+from testtools import (
+    get, docker_host, try_until, run, skip_if_go_version
+)
 
-command:
-    docker run -ti --volume-driver dvol -v hello:/data busybox sh
-expected output:
-    dvol volume is created on-demand
+DVOL = "/usr/local/bin/dvol"
 
-command:
-    dvol init hello2
-expected output:
-    Created volume hello2
-    Created branch hello2/master
+class VoluminousTests(TestCase):
+    def setUp(self):
+        self.tmpdir = FilePath(self.mktemp())
+        self.tmpdir.makedirs()
+
+    def test_docker_run_test_container(self):
+        def cleanup():
```

Comment:
The cleanup can be made a lot 'cleaner' using the 'fixtures' module. I won't leave it at that, I'll follow up with an example in a few minutes